### PR TITLE
Fix IEEE Fetcher by enabling cookie support

### LINKED ISSUE #3966: Problems finding fulltext for IEEExplore
Issues with IEEE full-text importer.

I have tested the latest master version from http://builds.jabref.org/master/JabRef--master--latest.jar on Windows 7 and the following problem is encountered.

JabRef version 4.2-dev on Windows 7 Enterprise (build 7601 service pack:1)

Steps to reproduce:

1. Use Jabref on a network with full-text access to IEEExplore articles (such as a university's network)
2. Start a new library with the following bibtex entry

```
@inproceedings{Morabito2017,
author = {Morabito, Bruno and Klein, Reinhardt and Findeisen, Rolf},
booktitle = {2017 American Control Conference (ACC)},
doi = {10.23919/ACC.2017.7963481},
isbn = {978-1-5090-5992-8},
month = {may},
pages = {3457--3462},
publisher = {IEEE},
title = {{Real time feasibility and performance of moving horizon estimation for Li-ion batteries based on first principles electrochemical models}},
url = {http://ieeexplore.ieee.org/document/7963481/},
year = {2017}
}
```

3.  Select this reference entry within Jabref and hit ```Alt``` + ```F7``` or go to ```Quality -> Look up full text documents```
4. After a while, the following pop-up dialog box opens
```No full text document found for entry Morabito2017```
5. Expected behavior: Navigate to this same ```URL http://ieeexplore.ieee.org/document/7963481/```.  This works perfectly fine and full-text PDF can be manually downloaded.  The auto downloader should have obtained this PDF and linked it to the citation entry.

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #3966 

Apparently when in VPN network of University IEEE fulltext fetcher needs cookie support enabled

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fetcher/IEEE.java [hunk 1]
```diff
@@ -45,12 +46,16 @@
             }
         }
 
+        //        12:50:19.221 GEThttps://ieeexplore.ieee.org/ielx7/6287639/7419931/07421926.pdf[HTTP/1.1 200 OK 8421ms]
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/logic/importer/fetcher/IEEE.java]: Remove these two lines

File: src/main/java/org/jabref/logic/importer/fetcher/IEEE.java [hunk 2]
```diff
@@ -66,7 +71,9 @@
         }
 
         // Download the HTML page containing a frame with the PDF
-        String framePage = new URLDownload(BASE_URL + stampString).asString();
+        URLDownload dl = new URLDownload(BASE_URL + stampString);
+        dl.getCookieFromUrl(); //We don't need to modify the cookies, but we need support for them
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/logic/importer/fetcher/IEEE.java]: Add comment in line before and add a space.

File: src/main/java/org/jabref/logic/importer/fetcher/IEEE.java [hunk 3]
```diff
@@ -45,12 +46,16 @@
...
+        //        12:50:19.221 GEThttps://ieeexplore.ieee.org/ielx7/6287639/7419931/07421926.pdf[HTTP/1.1 200 OK 8421ms]
+
         // If not, try DOI
         if (stampString.isEmpty()) {
             Optional<DOI> doi = entry.getField(FieldName.DOI).flatMap(DOI::parse);
             if (doi.isPresent() && doi.get().getDOI().startsWith(IEEE_DOI) && doi.get().getExternalURI().isPresent()) {
                 // Download the HTML page from IEEE
-                String resolvedDOIPage = new URLDownload(doi.get().getExternalURI().get().toURL()).asString();
+                URLDownload dl = new URLDownload(doi.get().getExternalURI().get().toURL());
+                dl.getCookieFromUrl();
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/logic/importer/fetcher/IEEE.java]: Also add the comment here please

File: src/main/java/org/jabref/logic/importer/fetcher/IEEE.java [hunk 4]
```diff
@@ -50,7 +51,11 @@
             Optional<DOI> doi = entry.getField(FieldName.DOI).flatMap(DOI::parse);
             if (doi.isPresent() && doi.get().getDOI().startsWith(IEEE_DOI) && doi.get().getExternalURI().isPresent()) {
                 // Download the HTML page from IEEE
-                String resolvedDOIPage = new URLDownload(doi.get().getExternalURI().get().toURL()).asString();
+                URLDownload dl = new URLDownload(doi.get().getExternalURI().get().toURL());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/importer/fetcher/IEEE.java]: please rename `dl` to something more meaningful

**Pull request merged**